### DISCUSSION TITLE
how to useStyleConfig correctly for ChakraUI component?

### DISCUSSION THREAD

**Discussion by zgid123** [chakra-ui/chakra-ui#7335] [Q&A]
Hi,

Assume I have a component that will use `input` and `text` from `chakra-ui`. I wanna apply  style from chakra theme for that, but the input and the text components will ignore the property `__css`.

```tsx
import { Box, Input, Text, useMultiStyleConfig } from '@chakra-ui/react';

export function MyComponent(): JSX.Element {
  const { label, input } = useMultiStyleConfig('MyComponent', {});

  return (
    <Box>
      <Input __css={input} />
     <Text __css={label}>Test Text</Text>
    </Box>
  );
}
```

 Usually we will do

```tsx
import { Box, chakra, useMultiStyleConfig } from '@chakra-ui/react';

export function MyComponent(): JSX.Element {
  const { label, input } = useMultiStyleConfig('MyComponent', {});

  return (
    <Box>
      <chakra.input __css={input} />
     <chakra.p __css={label}>Test Text</chakra.p>
    </Box>
  );
}
```

But I wanna apply style config of input from chakra-ui and only override some css properties. But if I use like this, typescript will raise error due to typing (1)

```tsx
import { Box, Input, Text, useMultiStyleConfig } from '@chakra-ui/react';

export function MyComponent(): JSX.Element {
  const { label, input } = useMultiStyleConfig('MyComponent', {});

  return (
    <Box>
      <Input {...input} />
      <Text {...label} >Test Text</Text>
    </Box>
  );
}
```

There is a work around (2)

```tsx
import { Box, chakra, useMultiStyleConfig, useStyleConfig } from '@chakra-ui/react';

export function MyComponent(): JSX.Element {
  const { label, input } = useMultiStyleConfig('MyComponent', {});
  const inputStyles = useStyleConfig('Input', {});
  const labelStyles = useStyleConfig('Text', {});

  return (
    <Box>
      <chakra.input __css={{ ...inputStyles, ...input }} />
     <chakra.p __css={{ ...labelStyles, ...label }}>Test Text</chakra.p>
    </Box>
  );
}
```

For solution 1, I can force casting typing for the property

```tsx
<Input {...input as InputProps} />
```

I wanna know what is the correct way to apply the styling for this case. I really don't wanna use solution 2. But if it's the correct way to apply styling for chakra-ui, then I will use it instead.

Also, if I use solution 1, is there any problem with that?

**Comment by keithburgie** [comment]
I'm fighting with something similar right now and looking through these questions for help 😆 Some better instructions for styling this way would definitely be appreciated.

For your case, try this:
```
<Box>
  <Box as={Input} __css={input} />
  <Box as={Text} __css={label}>Test Text</Text>
</Box>
```

**Reply by zgid123** [reply]
after using your solution awhile, I think I will use solution 2, because I don't wanna use `as` for this case 😂. Thanks for your solution
